src/components/ProgressIndicator/ProgressIndicator.tsx

```diff
@@ -79,6 +91,12 @@ export const ProgressIndicator: ChakraComponent<
       );
       finalValue = 0;
     }
+    if (indicatorType === "linear" && labelPlacement !== "bottom") {
+      console.warn(
+        "NYPL Reservoir Progress Indicator: The `labelPlacement` prop has been passed, " +
+          "but the `'linear'` `indicatorType` variant will not use it."
+      );
+    }
```

Comment:
I think we're going to need to do more for this error warning. Right now, if a consuming app passes both `linear` and `bottom`, no error message will show, leaving the possibility for confusion. What if you removed the default value for `labelPlacement` and created a new `finalLabelPlacement` prop where the value of `finalLabelPlacement` was set to something like this:

```
const finalLabelPlacement = !labelPlacement ? "bottom" : labelPlacement;
```

`finalLabelPlacement` could be passed to `useMultiStyleConfig` and `labelPlacement` could be used for the condition that controls the warning.

```
if (indicatorType === "linear" && labelPlacement) { ... }
```

Reply:
Should we just call the prop `circularIndicatorLabelPlacement`? It could also be a boolean instead that is `circularIndicatorLabelToRight` or something like that, since there is only one option other than the default.

Reply:
The requirements for the prop were updated, so now there are 4 options for placement bottom, top, right, and left.

Reply:
Oh I see, thank you @jackiequach! 

One more suggestion: rather than the console warning, what if we instead used a [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions) to make the type such that you could only pass the `labelPlacement` prop if the variant was `circular`?

Reply:
> Should we just call the prop `circularIndicatorLabelPlacement`? It could also be a boolean instead that is `circularIndicatorLabelToRight` or something like that, since there is only one option other than the default.

There is a chance that the label placement functionality is eventually added to the linear variant as well, so let's keep the prop name agnostic for now.

Reply:
Oh sorry, didn't see this. I will revert my change.

Reply:
I think the discriminated union is a great idea! It would flag devs at typecheck instead of in the browser.